### PR TITLE
[Woo POS] Add WooCommerce POS framework

### PR DESCRIPTION
**Pull request by joshheald**
## Description

This adds a new app-level framework for WooCommerce POS.

This is intended to sit at the same level in the heirarchy as the WooCommerce target, i.e. it can import Yosemite or an equivalent framework. 

This new framework should contain UI and VMs for the Woo POS project, with as much business logic as possible in Yosemite or an equivalent level framework, isolating the view from details of networking or storage type concerns.

### Testing with imported frameworks
I have (separately) tested this making a network request, and importing that lower-level framework works well. Note that's not the intended approach, it's just a lot simpler to test by skipping out Yosemite, while conceptually it proves the same thing.

To do this test, I passed in the `Credentials` and `siteID`, and used them to create an `OrdersRemote` and make a GET request with it.

## Testing instructions

1. Launch the app
2. Navigate to Menu > Settings > Experimental features
3. Turn on the Woo POS toggle
4. Switch stores or restart the app
5. Navigate to Menu > Woo POS
6. Observe that the placeholder view is shown

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/202">18.6</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Podfile [hunk 1]
```diff
@@ -120,6 +120,22 @@ target 'WooCommerce' do
   end
 end
 
+
```
Comment (dangermattic) [inline Podfile:123]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/woocommerce/woocommerce-ios/blob/f9fa900c88d33d5a68df84ba433f380165d6d36f/Podfile#L123"></span>Layout/EmptyLines: Extra blank line detected.</td>
    </tr>
  </tbody>
</table>

File: Podfile [hunk 2]
```diff
@@ -120,6 +120,31 @@ target 'WooCommerce' do
...
+# Unit Tests
+# ==========
+#
+target 'WooCommercePOSTests' do
+  project 'WooCommercePOS/WooCommercePOS.xcodeproj'
+  woocommercepos_pods
+end
+
+
```
Comment (dangermattic) [inline Podfile:147]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/woocommerce/woocommerce-ios/blob/6862030c1476bedec380e7e83f8ba81a1974786f/Podfile#L147"></span>Layout/EmptyLines: Extra blank line detected.</td>
    </tr>
  </tbody>
</table>

**Review by jaclync — APPROVED** [review]
LGTM, nice work getting the framework set up quickly!! Tested the debug build in iPhone/iPad simulators plus installable build on a physical device and the POS screen works as expected 🚀

File: WooCommercePOS/WooCommercePOS/Point Of Sale/PointOfSaleEntryPointView.swift [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
-    var body: some View {
+    public init(showFullScreen: Bool = true) {
+        self.showFullScreen = showFullScreen
+    }
+
+    public var body: some View {
         VStack {}
         .fullScreenCover(isPresented: $showFullScreen) {
-            if UIDevice.isPad() {
+            if UIDevice.current.userInterfaceIdiom == .pad {
```
Comment (iamgabrielma) [inline WooCommercePOS/WooCommercePOS/Point Of Sale/PointOfSaleEntryPointView.swift:15]: Ha! The exact case we made to catch hidden, implicit dependencies 💯

**Comment by iamgabrielma** [comment]
Builds and tests well! I found a building issue when trying to run the tests for the new framework, but I'm unsure if is expected at this point:

- Switch the scheme to `WooCommercePOS`
- In `WooCommercePOSTests` add a dummy test:
```
final class WooCommercePOSTests: XCTestCase {
    func test_if_foo() {
        XCTFail()
    }
}
```
- Attempt to run the test suite. Observe the build failed with error message pointing to denied file-write access:
```
Sandbox: rsync.samba(24382) deny(1) file-write-create /Users/iamgabrielma/Library/Developer/Xcode/DerivedData/WooCommerce-acxmclmulwdqerdohwsuwvzdrppn/Build/Products/Debug-iphonesimulator/WooCommercePOSTests.xctest/Frameworks/CocoaLumberjack.framework/CocoaLumberjackPrivacy.bundle
```
Screenshot 2024-05-09 at 10 49 40

From some research it would seem to come from the inclusion of `Xcode Build Script Sandboxing` since Xcode14, and a conflict between Xcode15 and cocoapods when the `ENABLE_USER_SCRIPT_SANDBOXING` build setting is enabled by default on those projects, which controls whether any “Run Script” build phases will be run in a sandbox or not. If I switch the setting to `No` in the build settings, the test target builds fine and we can run the tests:

```diff
- ENABLE_USER_SCRIPT_SANDBOXING = YES;
+ ENABLE_USER_SCRIPT_SANDBOXING = NO;
```

I believe this should be fine, but looping @mokagio in case we need to know of any other consideration about disabling this setting (the other targets do not have it enabled, so would be consistent), since he also participated on this thread: https://github.com/CocoaPods/CocoaPods/issues/11946

**Comment by jaclync** [comment]
> Builds and tests well! I found a building issue when trying to run the tests for the new framework, but I'm unsure if is expected at this point: ...
> From some research it would seem to come from the inclusion of Xcode Build Script Sandboxing since Xcode14, and a conflict between Xcode15 and cocoapods when the ENABLE_USER_SCRIPT_SANDBOXING build setting is enabled by default on those projects, which controls whether any “Run Script” build phases will be run in a sandbox or not. If I switch the setting to No in the build settings, the test target builds fine and we can run the tests:

Great catch on unit testing! This fixes the issue on my end. For reference, WooFoundation also has `ENABLE_USER_SCRIPT_SANDBOXING` set to `No` with CocoaLumberjack dependency as a pod.

Also, to include the POS tests from the framework, maybe we can add the POS tests in editing WooCommerce scheme > Test > (Test Plans) UnitTests, then add WooCommercePOSTests (https://github.com/woocommerce/woocommerce-ios/commit/d5974df64409e49587f9d1be24bd93301b360181). This would probably enable unit testing on CI for the new framework.

File: Podfile [hunk 4]
```diff
@@ -120,6 +120,29 @@ target 'WooCommerce' do
   end
 end
 
+# WooCommerce POS Layer:
+# ===============
+#
+def woocommercepos_pods
+  cocoa_lumberjack
```
Comment (jaclync) [inline Podfile:127]: I meant to ask earlier, is CocoaLumberjack included just for testing Cocoapod dependencies or for loggin like DDLog in the future?

Comment (joshheald) [reply Podfile:127]: It was a good simple import to check that the pods project setup was working properly, but also I figured that we'd definitely want DDLog, so we won't have to undo it.

**Comment by joshheald** [comment]
Thanks @jaclync and @iamgabrielma 

Changes for tests added in c2f17a1 – I'll merge this now as we have consensus on using it as a starting approach. If we find the modularisation efforts are taking too much time, we can merge back to the WooCommerce target easily, but I hope that won't be the route we have to take.

**Pull request merged**